pkg/kv/bulk/buffering_adder.go

```diff
@@ -287,18 +291,22 @@ func (b *BufferingAdder) doFlush(ctx context.Context, forSize bool) error {
 	}
 
 	if log.V(1) {
-		if b.sink.stats.span.EndKey.Compare(mvccKey.Key) < 0 {
-			b.sink.stats.span.EndKey = mvccKey.Key.Clone()
+		if b.sink.span.EndKey.Compare(mvccKey.Key) < 0 {
+			b.sink.span.EndKey = mvccKey.Key.Clone()
 		}
 	}
 
-	b.sink.stats.flushWait += timeutil.Since(beforeFlush)
+	b.sink.currentStats.FlushWait += timeutil.Since(beforeFlush)
 
 	if log.V(3) {
 		written := b.sink.mu.totalRows.DataSize - beforeSize
-		files := b.sink.stats.batches - before.batches
-		dueToSplits := b.sink.stats.batchesDueToRange - before.batchesDueToRange
-		dueToSize := b.sink.stats.batchesDueToRange - before.batchesDueToRange
+		afterStats := b.sink.mu.totalStats.Identity().(*backuppb.IngestionPerformanceStats)
+		afterStats.Combine(&b.sink.mu.totalStats)
```

Comment:
If beforeStats = totalStats + curStats before flush, and afterStats = totalStats + curStats after flush, isn't it adequate to do curStats after flush - curStats before flush?